index.html

```diff
@@ -631,13 +644,32 @@ <h1>Terminology</h1>
       </dt>
       <dd>The combination of Public Security Metadata, Private Security Data, and any other configuration
         information (such as public keys) necessary to operationally configure the security mechanisms of a Thing.</dd>
+      <dt>
+        <dfn>Service</dfn>
+      </dt>
+      <dd>A Service is a software entity that has a network interface.
+      Services can be described by a <a>Thing Description</a> and are a kind of <a>Thing</a>.
+      See also <a>Virtual Thing</a>.
+      Compare with <a>Device</a>.</dd>
       <dt>
         <dfn>Servient</dfn>
       </dt>
       <dd>A software stack that implements the WoT building
         blocks. A Servient can host and expose Things and/or host Consumers that consume Things.
         Servients can support multiple Protocol Bindings to enable
         interaction with different IoT platforms.</dd>
+      <dt>
+        <dfn>Shadow</dfn>
+      </dt>
+      <dd>A Shadow is a <a>Virtual Thing</a> that
+      maintains a copy of the state and mediates interactions with another
+      <a>Thing</a> which may not be immediately available due to network
+      connectivity, power savings modes, or other factors.
+      A Shadow aims to achive eventual consistency with the state
+      of the Thing it represents.
+      If a Shadow has more advanced functionality than simply caching state
```

Comment:
What do you mean by "advanced functionality"? Could this be interactions via other (= non-property) affordances? 
Is a event consumer still a shadow or is it already a twin?

I believe there's no clear border between shadows and twins, suggest to use these terms as synonyms.

Reply:
I also wasn't sure about the distinction between a Shadow and a Digital Twin either, but the distinction being one-way (physical device to shadow) vs. two-way (between physical device and twin) makes a lot of sense. I haven't done a comprehensive literature review but this does seem to match the definition used by organisations like [Fraunhofer IESE](https://www.iese.fraunhofer.de/en/services/digital-twin.html), [MDPI](https://www.mdpi.com/2075-5309/11/4/151/pdf) and [PSA](https://www.psa.inc/company/news/digital-model-vs-digital-shadow-vs-digital-twin-and-their-place-in-eiot/).

Reply:
I would prefer to have a Shadow JUST mirror state, and be specific about that.  It's a lot easier to reason about Shadows as a part of a system if that's all it does, e.g. for the case of a hub having to deal with client devices that need to sleep and just call in occasionally on their own schedule to sync up.  If it's a Shadow we can reason about it like a slow version of the actual device.  If it's a Digital Twin it could be doing anything.
 
We could make Shadow a special case of a Digital Twin but then it's unclear whether we JUST mean mirroring when we say Digital Twin.  I would prefer to reserve Digital Twin for cases where it does more than just mirroring.

Reply:
One more minor technical point: hubs may have Things that "represent" other devices that may not actually have Thing Descriptions, for example a Service in a hub may present a network interface and shadow some other client device without a Thing Description, like a Z Wave or Zigbee device.  Not sure how to deal with that right now, we may have to loosen the definitions slightly to allow for shadowing of entities that don't have actual Thing Descriptions and/or that Thing Descriptions can't currently handle.

Reply:
Re one-way vs two-way: I will have to look at the definitions you cite e.g. by Fraunhofer.  But to me the sync would go both ways.  Suppose I have a switch that for some weird reason is only a client and only syncs periodically.  The I want to "read" a binary property on the Shadow to know what state the switch is in, but I might also want to "write" the property to flip the state.  In this case the real switch (when it calls in) should check the state on the switch and sync with it.   So then the Shadow acts like a "slow" version of the real device, with a latency (at most) equal the interval at which the client device wakes up and checks in.

Anyway, I think they are different, if not for precisely this reason.